agent/consul/stream/event_publisher_test.go

```diff
@@ -233,7 +239,20 @@ func TestEventPublisher_SubscribeWithIndexNotZero_CanResume(t *testing.T) {
 
 	publisher := NewEventPublisher(newTestSnapshotHandlers(), time.Second)
 	go publisher.Run(ctx)
-	// Include the same event in the topicBuffer
+
+	// Start a separate subscription that remains open throughout the entire test
+	// to prevent the buffer getting cleaned up because there aren't any subscribers.
+	sub, err := publisher.Subscribe(req)
+	require.NoError(t, err)
+	defer sub.Unsubscribe()
+
+	// Publish the testSnapshotEvent, to ensure that it is skipped over when
+	// splicing the topic buffer onto the snapshot.
+	//
+	// Note: we must evict the snapshot that was created by the above subscription
+	// first, as it has already had the topic buffer spliced onto it (so therefore
+	// would include the snapshot event again).
+	delete(publisher.snapCache[req.Topic], req.TopicKey())
```

Comment:
This delete is racey and would fail the race detector right? I think you'd need to hold the publisher's lock if you want to play with its internals here. Ideally all our tests should pass with race detector enabled, but especially this package with it's creative concurrency patterns!

Reply:
The `stream` package does run with the race detector enabled in `go-test-race`. So if this is a race we should be able to reproduce with a test in that package.

Reply:
Great call out! 🐎